microovn/ovn/commands.go

```diff
@@ -0,0 +1,150 @@
+package ovn
+
+import (
+	"errors"
+	"fmt"
+	"github.com/canonical/microcluster/state"
+	"github.com/lxc/lxd/shared"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+const defaultDBConnectWait = 30 //Default time to wait for connection to ovsdb
+
+// ovsdbSpec is a helper structure for precise identification of ovsdb databases. A lot of
+// ovn/ovs commands take path to either database file, database socket or process control socket
+// along with the database name. This structure can be used for such cases.
+type ovsdbSpec struct {
+	Path string
```

Comment:
The use of the keyword `Path` does not resonate with me. The tools you are modelling can accept many types of connection methods which may be represented as paths, remote IP addresses or something else.

PS: This is just a command wrapper so we do not yet have a lot of requirements for it, but just wanted to make you aware of the existence of https://github.com/ovn-org/libovsdb

Reply:
> The use of the keyword Path does not resonate with me.

In the current context, the only "end-user" of this value was `ovsd-client wait` command intended to run against local target but you are right, the semantic should be improved. Perhaps we can rename it to `Target` and also rename the `OvsdbType` enum members from
```
OvsdbTypeNB
OvsdbTypeSB
```
to
```
OvsdbTypeNBlocal
OvsdbTypeSBlocal
```
to make it more explicit. And later when the need arises we can add new enum members with `remote` suffix and add logic for generating those value.

> PS: This is just a command wrapper so we do not yet have a lot of requirements for it, but just wanted to make you aware of the existence of https://github.com/ovn-org/libovsdb

I did not. thank you. This is good stuff, I'll look into whether it makes sense to start using it right now to remove some of the clunkyness. I'm also not a fan of invoking shell when it's not necessary.